ssl/t1_lib.c

```diff
@@ -3456,3 +3455,17 @@ size_t ssl_hmac_size(const SSL_HMAC *ctx)
     return 0;
 }
 
+int ssl_get_EC_curve_nid(const EVP_PKEY *pkey)
+{
+    int ret = NID_undef;
+    EVP_PKEY_CTX *ctx = NULL;
+    char gname[OSSL_MAX_NAME_SIZE];
+
+    if ((EVP_PKEY_is_a(pkey, "EC") || EVP_PKEY_is_a(pkey, "SM2"))
+        && ((ctx = EVP_PKEY_CTX_new((EVP_PKEY *)pkey, NULL)) != NULL
+            || EVP_PKEY_CTX_get_group_name(ctx, gname, sizeof(gname)) > 0))
```

Comment:
Don't we need to modify `EVP_PKEY_CTX_get_group_name` to work with legacy keys? Because it doesn't right now.

Reply:
Oops, forgot about that. Will fix tomorrow